### ISSUE TITLE
Integrate with Schema.org

### ISSUE THREAD

**Issue by rektide** [dtinit/data-transfer-project#306]
Looking at [PhotoModel](https://github.com/google/data-transfer-project/blob/master/portability-types-transfer/src/main/java/org/datatransferproject/types/transfer/models/photos/PhotoModel.java) for example, I don't see any existing schema or ontology that this newly created type relates to. This seems to be a brand new, unrelated, clean room schema. That seems like a colossal wasted opportunity to me, and could seriously, critically, massively set back the utility & capabilities of this project.

I would greatly like to see the types in Data-Transfer-Project linked, in a any cases at all possible, to the existing semantic web schemas that exist. So PhotoModel for example ought hopefully be implementable as a [http://schema.org/Photograph](http://schema.org/Photograph).

Mail should be derived from http://schema.org/EmailMessage
Task from http://schema.org/Action
Contacts are http://schema.org/ContactPoint
Calendar is a http://pending.schema.org/Schedule

Please please please, kindly make use of the good object definitions that already exist & that we have spent so much effort standardizing and supporting. This will make it much easier for consumers to make good use of this data, and it is critically important that this project take the responsibility of participating with the existing work that it overlaps so broadly with.

**Comment by danbri** [comment]
It would be good to use common schemas.  Speaking for Schema.org, let's balance any criticism with collaboration. Getting any kind of agreement on shared schemas is hard work, and the existing success here should be celebrated. If there is anything we can do at Schema.org to help (including tweaking or augmenting our schemas), that sounds like a good thing to collaborate on. We should also take care to distinguish the specific data shapes (e.g. with notions of mandatory/expected/recommended fields, rules about where to expect extensions) from the underlying definitions. Schema.org is much more in the latter business, although there are recent technologies like ShEx and SHACL that can be used to profile it down for definition of more rigid and predictable (eg. OO-like) data structures. We could also look at mappings rather than expecting Schema.org in JSON-LD, Microdata or RDFa be used directly...

**Comment by csuwildcat** [comment]
I think a fundamental point to deeply consider in this general pursuit is that mapping, transforms, scoping, etc., are the reason why data exchange protocols fail - you win this game by *not playing*. To illustrate: if I want product data from another identity, don't try forcing the world to all use one schema, transform, or agreed scope, instead allow all data schemas/types to be *directly addressable*, so the consuming entity asks for access to the 3 or 4 object types (across N schemas) that comprise the need. This is an organic approach that provides *deterministic, inferential knowability* - basically, calling entities implicitly know how to ask for what they need without any grand bargains or high coordination activities.

**Comment by bobdc** [comment]
It's a little late to make the argument that because overarching schemas usually fail and schema.org aims to be an overarching schema, schema.org is doomed to fail; it's already a huge success. Also, I don't think that the only alternative to One True Schema is Everyone Just Go Ahead and Make Up Your Own. I appreciate that building interoperability mechanisms right into an architecture is worth doing, but when a solid metadata standard for something in the Data Transfer Project domain is available and has good traction (e.g. the IPTC's photo metadata standard) it's at least worth some examination.

**Comment by csuwildcat** [comment]
@bobdc I think you're misunderstanding me: I'm absolutely saying you should use all the existing schemas, including schema.org. I'm talking about the mechanism of enabling their use/access without mapping, transforms, or scope agreements.

**Comment by danbri** [comment]
I'm all for file / mime level interop, fwiw. 

Schema.org's success stems in large part from mapping.

A vast number of Web sites are defined by mappings; templates in some format that generates HTML/CSS/JS from whatever the underlying representation is (often a database of some kind, e.g. SQL). Schema.org is very rarely used as the master format. Instead it rides along on the existing mapping infrastructure, you just include useful bits of Schema.org data in the output Web content. What you do in the privacy of your own database remains your own business, and the two can evolve at their own rhythms in response to changing circumstances. Something similar may work here, which is why I mentioned mappings.

**Comment by csuwildcat** [comment]
Again, I feel you're misunderstanding me:

Schema.org is great, and one of many great schemas, like GS1, FHIR, Lime, etc. By embracing a datastore based on inferential knowability, you can directly, deterministically address all of these at once, without writing a single scope, transform, or mapping layer - let the data *itself* be the API, get what I'm saying?

**Comment by akuckartz** [comment]
What is "inferential knowability" ?

**Comment by csuwildcat** [comment]
Inferential/implicit knowability is the idea that you specifically *do not*
attempt to map, transform, or scope unify 'all the things', but instead
letting the schema origin and data type pairs themselves be directly
addressable as the API surface. Wanna get data from an entity of a certain
kind? Just ask for the relevant schema/object types directly (assuming
you've created an interface/permissioning mechanisms similar to this:
https://github.com/decentralized-identity/hubs/blob/master/explainer.md#collections
).

In short: it's the high coordination approaches to addressing themselves
that causes the failure in data exchange systems, so just side step it and
win the game by not playing. (sounds counterintuitive at first, I know)

**Comment by akuckartz** [comment]
So I have to "ask for the relevant schema/object types directly". In the example, how do I know which are relevant in addition to http://schema.org/Offer ?

**Comment by csuwildcat** [comment]
@akuckartz it turns out that most major industries already use/adhere to a few well-know schemas, for example: FHIR (widely used medical schema), GS1 (basically the only supply chain schema), etc. App/service companies and devs use these daily, so they know which schemas/objects they're looking for. The headline: we can do this organically without federation systems.

**Comment by akuckartz** [comment]
I see. But why not simply use a Linked Data Platform (https://www.w3.org/TR/ldp/) ?

**Comment by joelhandwell** [comment]
@danbri, @RichardWallis, @vholland, @mfhepp any input?

**Comment by danbri** [comment]
@joelhandwell I left some notes above. My main concern (to be slightly blunt) is that we don't have a load of well meaning Semantic Web / Linked Data advocates jump in here lobbying on behalf of Schema.org. That mode of interaction, historically, tends not to go well. Repeatedly. We should take the data-transfer-project initiative for what it is and work towards common ground, they have some existing work, important traction amongst competing parties, and it's unrealistic to expect the project to start over from scratch. I don't mean this as a criticism of anyone in particular, just a red flag based on previous experience (including the launch of Schema.org).

Q for @csuwildcat.  What commonalities should we assume each data flavour shares with others? For  Bits/bytes? Being made of unicode? Can we go deeper, as e.g. XML, JSON, RDF, Protobuf etc tried to? 
You mention FHIR, GS1, alongside Schema.org. All of these have something like a graph data model; all broadly in the RDF family, e.g. FHIR has [ShEx schemas](https://www.hl7.org/fhir/implementationguide.shex.html) to define data shapes, the [GS1 Web vocabulary](http://gs1.org/voc/) was designed to work with Schema.org and JSON-LD, and Schema.org is also [based on](http://schema.org/docs/datamodel.html) entity-property-value triples. 

We may do better if we separate questions of shared approach to data structure, from matters of common descriptive vocabulary (specific types and properties, for example, or more complex shapes built from them).
